Geometry/HGCalCommonData/src/HGCalGeomTools.cc

```diff
@@ -33,7 +33,7 @@ void HGCalGeomTools::radius(double zf,
   auto zb1 = std::lower_bound(zFront1.begin(), zFront1.end(), zb);
   if (zb1 != zFront1.begin())
     --zb1;
-  if (std::abs(*zb1 - zb) < tol_) {
+  if (((zb1 - 1) != zFront1.begin()) && (std::abs(*zb1 - zb) < tol_)) {
```

Comment:
Shouldn’t the check be `zb1 != zFront1.begin()` since that is the case where you can not do `—zb1` ?